File: EmojibleIOS/EmojibleIOS/Models/ClassModel.swift
//
//  ClassModel.swift
//  EmojibleIOS
//
//  Created by Haldun Balim on 12.11.2020.
//

import Foundation
class ClassModel: Equatable{
    var id:String
    var teacherId:String
    var className:String
    var password:String
    
    init(id:String, teacherId:String, className:String, password:String) {
        self.id = id
        self.teacherId = teacherId
        self.className = className
        self.password = password
    }
    
    var dictionary: [String:Any]{
        return [
            "id":id,
            "teacherId":teacherId,
            "className": className,
            "password":password,
        ]
    }
    
    init (dictionary: [String:Any]) {
        id = dictionary["id"] as! String
        teacherId = dictionary["teacherId"] as! String
        className = dictionary["className"] as! String
        password = dictionary["password"] as! String
    }
    
    static func == (lhs: ClassModel, rhs: ClassModel) -> Bool {
        return lhs.id == rhs.id && lhs.teacherId == rhs.teacherId
    }
}

File: EmojibleAndroid/app/src/main/java/com/dji/emojibleandroid/models/ClassModel.kt
package com.dji.emojibleandroid.models

data class ClassModel(
    var id: String,
    var teacherId: String,
    var className: String,
    var password: String
) {
    var dictionary = hashMapOf<String, Any>(
        "id" to id,
        "teacherId" to teacherId,
        "className" to className,
        "password" to password
    )

    constructor(dict: HashMap<String, Any>) : this(
        dict["id"] as String,
        dict["teacherId"] as String,
        dict["className"] as String,
        dict["password"] as String
    )

    override fun equals(other: Any?): Boolean {
        if (other is ClassModel) {
            return this.id == other.id && this.teacherId == other.teacherId
        }
        return false
    }
}
